[UI] Welcome modal update

biliesilva opened the pull request:
<--- Update Welcome Modal --->

Updated the copy and some styles of the Welcome to HyperPlay modal

[Design to compare here](https://www.figma.com/design/oyHp4NcziMufK4t8wMCeTX/Onboarding?node-id=261-4538&t=ldJd23TmYU1KYcFx-1)


---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

biliesilva commented:
I have read the CLA Document and I hereby sign the CLA

flavioislima commented on public/locales/en/translation.json:
```diff
@@ -436,13 +435,18 @@
+                    "bulletPoint_4": "Install Steam (Windows version) on macOS with one easy click",
+                    "bulletPoint_5": "Run Windows games on macOS, Linux, and SteamDeck using our\n              advanced",
+                    "bulletTitle": "With HyperPlay, you can:",
+                    "communityCTA_4": "Let's shape the future of gaming together!",
```
The i18n workflow error is related with this line here. Running locally it gets removed since it is not used anywhere.
So try again to run `yarn i18n` and commit the change.

flavioislima commented on src/frontend/screens/Onboarding/welcome/index.tsx:
```diff
@@ -18,61 +19,96 @@ const Welcome: React.FC<WelcomeProps> = function (props) {
           {t(
-            'hyperplay.onboarding.welcome.text.communityCTA_1',
-            `We'd love your feedback and to have you join us in our`
+            'hyperplay.onboarding.welcome.text.2',
```
for this one, can you not use a number as a key, this might give issues in the future. 
Rename it to something like `intro` for instance.

flavioislima commented on src/frontend/screens/Onboarding/welcome/index.tsx:
```diff
@@ -18,61 +19,96 @@ const Welcome: React.FC<WelcomeProps> = function (props) {
             }
           >
-            {t('hyperplay.discordApp', `Discord`)}
+            {t('hyperplay.discordApp2', `Discord Community.`)}
```
I think this one could be named `discordCommunity` instead.

flavioislima reviewed (approved):
Pretty good stuff! 💪🏽 
Code overall looks good, just left some comments about the i18n keys.

BrettCleary commented:
can we take the time to make this modal responsive for smaller screens? You will also need to set the z-index as well or convert it to a `dialog` element with modal behavior, so it is above the top nav bar. Probably set a max height of `100vh` and y-overflow to `auto` as well

<img width="619" alt="image" src="https://github.com/user-attachments/assets/9f8d1d1f-8a40-4320-b31a-ec2f310589a8">
<img width="513" alt="image" src="https://github.com/user-attachments/assets/db06475e-a63b-4730-85c9-d05b8253d4ad">

BrettCleary commented on src/frontend/screens/Onboarding/welcome/index.module.scss:
```diff
@@ -32,8 +39,38 @@
 .welcomeBodyTextContainer a:hover {
   color: var(--color-primary-300);
   transition: 0.2s;
+  font-size: var(--font-size-md);
```
Do we need this? It seems odd to me that we would set the font size instead of using a typography style, especially on hover

biliesilva replied:
This was already in the file, I only changed it from lg to md;

BrettCleary commented on src/frontend/screens/Onboarding/welcome/index.module.scss:
```diff
@@ -32,8 +39,38 @@
+.diamondBlue {
+  width: 10px;
+  height: 10px;
+  margin-right: 8px;
```
```suggestion
  margin-right: var(--space-xs-fixed);
```

BrettCleary commented on src/frontend/screens/Onboarding/welcome/index.module.scss:
```diff
@@ -32,8 +39,38 @@
+  font-size: var(--font-size-md);
+}
+
+.text {
+  font-weight: 400;
 }
```
We pretty much never have to set the specific typography in a component since we can reuse the styles from hp/ui `src\styles\designSystem\_typography.scss`. Do we need a new style or to update an existing style in the global stylesheet? If we make custom type sizing, weights, fonts, or line-heights on a per component basis, the typography system quickly become unmaintainable.

biliesilva replied:
I had to create a class to style a small piece of text differently because it was inheriting bold from its parent. I'll see how to change it.

BrettCleary commented on src/frontend/screens/Onboarding/welcome/index.module.scss:
```diff
@@ -32,8 +39,38 @@
+  align-items: flex-start;
+}
+
+.scrollableModal {
```
this class doesn't seem to be used anywhere

BrettCleary reviewed (changes requested):
ui is looking good. let's take the time to make it responsive since this is the first impression on new users

biliesilva commented:
> ui is looking good. let's take the time to make it responsive since this is the first impression on new users

I didn't change anything in the responsiveness of this component I just changed the text and changed the position of the language selector and the button. I believe it is being imported from another component

BrettCleary reviewed (approved):
LGTM

nyghtstalker reviewed (approved):
Looks great! Love the new welcome modal! Great job! 

Changing the language is currently not working because it is a new text and has not yet been translated. This will be corrected at a later date.

![image](https://github.com/user-attachments/assets/e153da3f-2f3d-4c67-9f6b-9eed0a2cf72d)
![image](https://github.com/user-attachments/assets/1df50fe6-3baa-490c-b9aa-97c96bb9a30f)
